ofrak_core/ofrak/core/strings.py

```diff
@@ -77,3 +86,71 @@ async def modify(self, resource: Resource, config: StringFindReplaceConfig) -> N
         offsets = [m.start() for m in re.finditer(to_find, original_data)]
         for offset in offsets:
             await resource.run(BinaryPatchModifier, BinaryPatchConfig(offset, replace_with))
+
+
+@dataclass
+class AsciiString(ResourceView):
+    """
+    Resource representing a C-style null-terminated string of ASCII characters. The `text` string
+    is not null-terminated.
+    """
+
+    text: str
+
+    @index
+    def Text(self) -> str:
+        return self.text
+
+    @classmethod
+    def caption(cls, all_attributes) -> str:
+        try:
+            s = all_attributes[AttributesType[AsciiString]]
+            return f"string: '{s.text}'"
+        except KeyError:
+            return super().caption(all_attributes)
+
+
+class AsciiStringAnalyzer(Analyzer[None, AsciiString]):
+    """
+    Extract the decoded string from a C-style null-terminated string of ASCII characters.
+    """
+
+    targets = (AsciiString,)
+    outputs = (AsciiString,)
+
+    async def analyze(self, resource: Resource, config: None) -> AsciiString:
+        raw_without_null_byte = (await resource.get_data())[:-1]
+        return AsciiString(raw_without_null_byte.decode("ascii"))
+
+
+class StringsUnpacker(Unpacker[None]):
+    """
+    Unpack null-terminated strings of printable ASCII characters.
+    """
+
+    targets = (ProgramSection,)  # TODO: Other reasonable targets?
+    children = (AsciiString,)
+
+    async def unpack(self, resource: Resource, config: None) -> None:
+        if resource.get_data_id() is None:
+            return
+        if resource.has_tag(CodeRegion):
+            # code is less likely to have strings so more likely to have false positives
+            min_length = 8
+        else:
+            min_length = 2
+
+        # match sequences of printable characters of at least `min_length` ending with null byte
+        # printable characters defined as: ASCII between ' ' and '~', tab, newline, carriage return
+        pattern = rb"([ -~,\n,\t,\r]{%d,})\x00" % min_length
```

Comment:
Two comments on this:

- Usually, you don't need commas to separate characters in the character class brackets – this regular expression will match literal commas (see suggested fix below)
- This should probably be compiled with [`re.compile`](https://docs.python.org/3/library/re.html#re.compile) and stored as a class attribute, instead of rebuilding it every time the unpacker is run – this can have significant speed implications if this runs on lots of resources

```suggestion
        pattern = rb"([ -~\n\t\r]{%d,})\x00" % min_length
```

Reply:
Yeah compiling is a good idea. I suppose we'll need two compiled versions for the two possible `min_length` values.

Reply:
I wonder if it makes sense to have a class attribute that caches compiled versions. Then, it compiles on the first run of `unpack`, but has the regular expression cached for later. Or maybe, a method with `@lru_cache` decorator to return a compiled regular expression given the minimum number of characters?

Either way, instead of having two versions, I like the idea of generalizing. That way, if we want to make an optional component config for this to let the user control the minimum size, the code will not need to be refactored.

Reply:
I don't think some sort of caching is called for, if we want to add an optional config for minimum length we can just compile the appropriate pattern on the fly when a config is given.